clientconn.go

```diff
@@ -411,6 +433,9 @@ func (cc *ClientConn) Close() error {
 	}
 	cc.state = Shutdown
 	cc.stateCV.Broadcast()
+	if cc.events != nil {
+		cc.events.Finish()
```

Comment:
Yeah, github review UI is horrible ... :)

Don't we need a mutex to guard cc.events and s.events (because you reset its value to nil now)? I think it can align with cc.state to share the same lock then.